coniferest/evaluator.py

```diff
@@ -52,10 +58,16 @@ def combine_selectors(cls, selectors_list):
         indices[0] = 0
         indices[1:] = np.add.accumulate(lens)
 
+        # Assign a unique sequential index to every leaf
+        # The index is used for weighted scores
+        leaf_mask = selectors['feature'] < 0
+        leaf_count = np.count_nonzero(leaf_mask)
+        selectors[leaf_mask]['left'] = np.arange(0, leaf_count)
+
         for i in range(len(selectors_list)):
             selectors[indices[i]:indices[i + 1]] = selectors_list[i]
 
```

Comment:
This should be:
```python
for i in range(len(selectors_list=):
    selectors[indices[i]:indices[i + 1]] = selectors_list[i]

# Assign a unique sequential index to every leaf
# The index is used for weighted scores
leaf_mask = selectors['feature'] < 0
leaf_count = np.count_nonzero(leaf_mask)
selectors['left'][leaf_mask] = np.arange(0, leaf_count)
```
1. `leaf_mask`-related stuff should go after `selectors` initialization
2. `selectors[leaf_mask]['left'] = ` doesn't assign anything